### DISCUSSION TITLE
vici: handle IKEv1 reauthentication

### DISCUSSION THREAD

**Discussion by d1mg** [strongswan/strongswan#1921] [Q&A]
Hello!
I create an application with vici to monitor ipsec connections.
I have problems with handling the reauthentication for IKEv1 correctly. The documentation for IKEv1 says: 

> IKEv1 SAs are also rekeyed/reauthenticated using a make-before-break scheme. However only the IKE SA is affected. IPsec SAs are adopted by the new IKE SA and not recreated.

How or when should the new IKE SA adopte the children? What event and key should I choose to determine from which IKE SA to which IKE SA I should move the children?
Should I search for older IKE SAs with same name at the ike-updown event, when the new IKE SA comes up?
Should I move all childs to the newest IKE SA when the old IKE SA goes down at the ike-updown event?

Kind regards

**Comment by tobiasbrunner** [comment]
The [children_migrate](https://github.com/strongswan/strongswan/blob/master/src/libcharon/bus/listeners/listener.h#L239) event is currently not exposed via vici (I also don't really plan to change that). I guess you could implement some heuristic via existing events. But my advice: [stop using IKEv1](https://datatracker.ietf.org/doc/html/rfc9395)!
